applications/Preprocessor/lib/centaur.cpp

```diff
@@ -257,314 +205,160 @@ CentaurReader::CentaurReader(std::string filename) {
-               position->second[0] != index) {
-            position = boundaryConnections.find(++index);
-            for (auto& value : coordinate) {
-                currentLine >> value;
-            }
-            if (--remainderThisLine == 0) {
-                remainderThisLine = nodesOnLine;
-                currentLine = readLine();
-            }
-        }
-        next.coordinates.resize(1);
-        for (auto& value : coordinate) {
-            currentLine >> value;
-        }
-        next.coordinates[0] = coordinate;
-        remainderThisLine--;
-        index++;
-        if (position != boundaryConnections.end() &&
-            position->second.size() > 1) {
-            std::istringstream::pos_type currentFilePosition =
-                centaurFile.tellg();
-            auto oldRemainder = remainderThisLine;
-            std::istringstream::pos_type currentLinePosition =
-                currentLine.tellg();
-            auto oldLine = std::move(currentLine);
-            currentLine = std::istringstream(oldLine.str());
-            currentLine.seekg(currentLinePosition);
-            for (auto offset :
-                 Range<int>{static_cast<int>(position->second[1]) -
-                                static_cast<int>(position->second[0]),
-                            difference_generator(position->second),
-                            position->second.size() - 1}) {
-                logger.assert_always(offset > 0, "Need to skip % positions",
-                                     offset);
-                while (static_cast<uint32_t>(offset) > remainderThisLine) {
-                    currentLine = readLine();
-                    offset -= remainderThisLine;
-                    remainderThisLine = nodesOnLine;
-                }
-                while (offset-- > 0) {
-                    --remainderThisLine;
-                    for (auto& value : coordinate) {
-                        currentLine >> value;
-                    }
-                }
-                next.coordinates.push_back(coordinate);
-            }
-            currentLine = std::move(oldLine);
-            centaurFile.seekg(currentFilePosition);
-            remainderThisLine = oldRemainder;
-        }
-        if (HPGEM_LOGLEVEL >= Log::VERBOSE) {
-            for (auto outputCoordinate : next.coordinates) {
-                std::cout << "(" << outputCoordinate[0];
-                for (std::size_t i = 1; i < outputCoordinate.size(); ++i) {
-                    std::cout << ", " << outputCoordinate[i];
-                }
-                std::cout << ")" << std::endl;
-            }
-        }
-        logger(VERBOSE, "");
-    };
-    MeshSource::Node node;
-    increment(node);
-    return Range<MeshSource::Node>{node, std::move(increment), numberOfNodes};
+    centaurFile.readRawRecord(headerSize, rawInput.data());
+
+    std::uint32_t offset = 0;
+    memcpy(&version, rawInput.data(), sizeof(offset));
```

Comment:
nice

Reply:
Yes changed it following your previous comment. Problem with the reinterpret cast was that some of its uses indeed were undefined behaviour, but that the common implementations did do the desired thing.

The commit says that `type*` -> `char*` and then dereferencing is allowed, while `char*` -> `type*` and then dereferencing is not allowed (unless you of course got the `char*` from a `type*`).